tests/next_tests/integration_tests/feature_tests/cases.py

```diff
@@ -0,0 +1,358 @@
+        return np.ones(np.prod(shape), dtype=dtype) * 5
+    return np.dtype(dtype).type(5)
+
+
+@dataclasses.dataclass(frozen=True)
+class Builder:
+    partial: functools.partial
+
+    def build(self, *args, **kwargs) -> Self:
+        return self.partial(*args, **kwargs)
+
+    def __call__(self, *args, **kwargs) -> Any:
+        return self.build(*args, **kwargs)
+
+
+def make_builder(func):
+    """Create a fluid inteface for a function with many arguments."""
+
+    def make_setter(argname):
+        def setter(self, arg):
+            return self.__class__(
+                partial=functools.partial(
+                    self.partial, *self.partial.args, **(self.partial.keywords | {argname: arg})
+                )
+            )
+
+        setter.__name__ = argname
+        return setter
+
+    argspec = inspect.getfullargspec(func)
+
+    @dataclasses.dataclass(frozen=True)
+    class NewBuilder(Builder):
+        for argname in argspec.args + argspec.kwonlyargs:
+            locals()[argname] = make_setter(argname)
+
+    func_snake_words = func.__name__.split("_")
+    func_camel_name = "".join(word.capitalize() for word in func_snake_words)
+
+    NewBuilder.__name__ = f"{func_camel_name}Builder"
+
+    return NewBuilder(functools.partial(func))
+
+
+def allocate(
+    case: Case,
+    fieldview_prog: decorator.FieldOperator | decorator.Program,
+    name: str,
+) -> Builder:
+    """Allocate a parameter or return value from a fieldview code object with a fluid interface."""
+    return (
+        make_builder(allocate_)
+        .sizes(case.default_sizes)
+        .case(case)
+        .fieldview_prog(fieldview_prog)
+        .name(name)
+    )
+
+
+def get_param_types(
+    fieldview_prog: decorator.FieldOperator | decorator.Program,
+) -> dict[str, ts.FieldType | ts.TupleType | ts.ScalarType]:
+    if fieldview_prog.definition is None:
+        raise ValueError(
+            f"test cases do not support {type(fieldview_prog)} with empty .definition attribute (as you would get from .as_program())!"
+        )
+    annotations = typing.get_type_hints(fieldview_prog.definition)
+    return {
+        name: type_translation.from_type_hint(type_hint) for name, type_hint in annotations.items()
+    }
+
+
+def extend_sizes(
+    sizes: dict[common.Dimension, str],
+    extend: Optional[dict[common.Dimension, tuple[int, int]]] = None,
+) -> dict[common.Dimension, str]:
+    sizes = sizes.copy()
+    if extend:
+        for dim, (lower, upper) in extend.items():
+            sizes[dim] += upper - lower
+    return sizes
+
+
+RETURN = "return"
+
+
+def allocate_(
+    case: Case,
+    fieldview_prog: decorator.FieldOperator | decorator.Program,
+    name: str,
+    sizes: dict[common.Dimension, int],
+    strategy: Optional[FieldInitializer | ScalarInitializer] = None,
+    dtype: Optional[str] = None,
+    extend: Optional[dict[common.Dimension, tuple[int, int]]] = None,
+) -> common.Field | tuple[common.Field, ...] | int | float:
```

Comment:
For coherency with previous suggestions about scalar value types:
```suggestion
) -> common.Field | tuple[common.Field, ...] | ScalarValue:
```